app/models/loan.rb

```diff
@@ -210,4 +210,12 @@ def healthy?
   def health_status_available?
     return !loan_health_check.nil?
   end
+
+  def principal_balance
+    transactions.by_date.last.principal_balance
```

Comment:
You may want to do a `.try(:principal_balance)` as sometimes `.last` will return `nil` -- Same for `interest_balance` below.